Microsoft.Toolkit.Uwp.UI.Controls/AdMediator/UwpAdMediator.xaml.cs

```diff
@@ -0,0 +1,377 @@
+            myAdRefreshTimer.Tick += MyAdRefreshTimer_Tick;
+            myAdRefreshTimer.Start();
+        }
+
+        private int GetAdDuplexWeight()
+        {
+            return 0;
+
+            //// TODO: Change this logic to fit your needs.
+            //// This example uses Microsoft ads first in Canada and Mexico, then
+            //// AdDuplex as fallback. In France, AdDuplex is first. In other regions,
+            //// this example uses a weighted average approach, with 50% to AdDuplex.
+
+            // int returnValue = 0;
+            // switch (GlobalizationPreferences.HomeGeographicRegion)
+            // {
+            //    case "CA":
+            //    case "MX":
+            //        returnValue = 0;
+            //        break;
+            //    case "FR":
+            //        returnValue = 100;
+            //        break;
+            //    default:
+            //        returnValue = 50;
+            //        break;
+            // }
+            // return returnValue;
+        }
+
+        private void ActivateMicrosoftBanner()
+        {
+            // Return if you hit the error limit for this refresh interval.
+            if (errorCountCurrentRefresh >= this.MaxErrorsPerRefresh)
+            {
+                myAdGrid.Visibility = Visibility.Collapsed;
+                return;
+            }
+
+            //// Use random number generator and house ads weight to determine whether
+            //// to use paid ads or house ads. Paid is the default. You could alternatively
+            //// write a method similar to GetAdDuplexWeight and override by region.
+            // string myAdUnit = myMicrosoftPaidUnitId;
+            // int houseWeight = HOUSE_AD_WEIGHT;
+            // int randomInt = randomGenerator.Next(0, 100);
+            // if (randomInt < houseWeight)
+            // {
+            //    myAdUnit = myMicrosoftHouseUnitId;
+            // }
+
+            // Hide the AdDuplex control if it is showing.
+            if (myAdDuplexBanner != null)
+            {
+                myAdDuplexBanner.Visibility = Visibility.Collapsed;
+            }
+
+            // Initialize or display the Microsoft control.
+            if (myMicrosoftBanner == null)
+            {
+                myMicrosoftBanner = new AdControl();
+                myMicrosoftBanner.ApplicationId = this.MicrosoftAppId;
+                myMicrosoftBanner.AdUnitId = this.MicrosoftAdUnit;
+                myMicrosoftBanner.Width = this.AdWidth;
+                myMicrosoftBanner.Height = this.AdHeight;
+                myMicrosoftBanner.IsAutoRefreshEnabled = false;
+
+                myMicrosoftBanner.AdRefreshed += MyMicrosoftBanner_AdRefreshed;
+                myMicrosoftBanner.ErrorOccurred += MyMicrosoftBanner_ErrorOccurred;
+
+                myAdGrid.Children.Add(myMicrosoftBanner);
+            }
+            else
+            {
+                myMicrosoftBanner.Visibility = Visibility.Visible;
```

Comment:
Maybe instead of setting visibility of the ad units everywhere change visual state and create visual states that will change visibility